extensions/amp-story/1.0/amp-story.js

```diff
@@ -1148,6 +1155,22 @@ export class AmpStory extends AMP.BaseElement {
           this.getPageIndex(oldPage) < pageIndex ?
             setAttributeInMutate(oldPage, Attributes.VISITED) :
             removeAttributeInMutate(oldPage, Attributes.VISITED);
+
+          if (oldPage.isAd()) {
+            this.storeService_.dispatch(
+                Action.SET_ADVANCEMENT_MODE,
+                AdvancementMode.ADVANCE_TO_ADS);
+          } else {
+            const advanceAttr =
+              isExperimentOn(this.win, 'amp-story-branching') ?
+                Attributes.PUBLIC_ADVANCE_TO :
+                Attributes.ADVANCE_TO;
+            if (oldPage.element.hasAttribute(advanceAttr)) {
+              this.storeService_.dispatch(
+                  Action.SET_ADVANCEMENT_MODE,
+                  AdvancementMode.ADVANCE_TO);
```

Comment:
Just curious about the product decision: Is this overriding any advancement mode previously set?
For example, if the advancement was media based, are we ok with `AUTO_ADVANCE_MEDIA` being overriden by `ADVANCE_TO`?

Reply:
Hmm, actually, good catch.  Now that I think of it, I'm not sure `ADVANCE_TO` is actually an advancement "mode," per se.  It would still be a manual advancement, just to the page specified by the `advance-to` attribute.

But, they already know what page the advancement was to, from the page ID/index.  So, I'm not sure knowing that it was because of the `advance-to` attribute tells them any new information?

Reply:
Good point. The new page id and old page id are available, so it'd make sense to keep the "manual", "media based", or "time based" information.